compiler/rustc_codegen_llvm/src/coverageinfo/mapgen.rs

```diff
@@ -26,6 +26,11 @@ use tracing::debug;
 /// undocumented details in Clang's implementation (that may or may not be important) were also
 /// replicated for Rust's Coverage Map.
 pub fn finalize<'ll, 'tcx>(cx: &CodegenCx<'ll, 'tcx>) {
+    // Ensure LLVM supports Coverage Map Version 4 (encoded as a zero-based value: 3).
+    // If not, the LLVM Version must be less than 11.
+    let version = coverageinfo::mapping_version();
+    assert_eq!(version, 3, "rustc option `-Z instrument-coverage` requires LLVM 11 or higher.");
```

Comment:
Apologies, I wasn't super clear in my last comment. We shouldn't `assert` because that will ICE the compiler, we just want a regular error message. 

```suggestion
    if version != 4 {
        cx.tcx.sess.fatal("rustc option `-Z instrument-coverage` requires LLVM 11 or higher.");
    }
```

Reply:
Ah, OK. I don't think I've had to use this API in the coverage map code before so I wasn't aware of that practice. Thanks.

Reply:
This needs to be, `if version != 3` though. (zero-based)